### PR TITLE
Fix double render in org signup

### LINKED ISSUE #6721: DoubleRenderError at some organization signups
Some organization signup pages are failing: https://rollbar.com/vizzuality/CartoDB/items/13737/?item_page=0&#instances

1.- Check why.
2.- In those cases where failing is legit, avoid the 500 error because of `AbstractController::DoubleRenderError` (it should just be a 404).

### PR DESCRIPTION
**Pull request by javitonino**
Fixes a Double Render that happened when displyaing signup for an organization that had signups disabled and the quota was exceeded. Reorganized the code to check if signup is enabled first.

Closes #6721 
CR @juanignaciosl

### CODE REVIEW

File: app/controllers/signup_controller.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@ class SignupController < ApplicationController
 
   skip_before_filter :http_header_authentication, only: [:create_http_authentication]
 
-  before_filter :load_organization, only: [:create_http_authentication]
+  before_filter :check_organization_quotas, only: [:create_http_authentication]
```
Comment (juanignaciosl) [inline app/controllers/signup_controller.rb]: Refactor is nice, but I'd suggest one more step forward: `check_organization_quotas` now loads the organization, which is a concern that it shouldn't have. You can add another `before_filter` for `load_organization` for `create_http_authentication` so you can remove `load_organization` from check_organization_quotas` and then remove the "already loaded check" from `load_organization.

**Comment by javitonino** [comment]
CR @juanignaciosl

**Comment by juanignaciosl** [comment]
:+1: (assuming it's tested ;-) ).

**Comment by javitonino** [comment]
retest this

**Comment by javitonino** [comment]
retest this, pleas

**Comment by javitonino** [comment]
retest this, please

**Pull request merged**